lib/pure/collections/sets.nim

```diff
@@ -227,6 +224,23 @@ proc toHashSet*[A](keys: openArray[A]): HashSet[A] =
   result = initHashSet[A](rightSize(keys.len))
   for key in items(keys): result.incl(key)
 
+proc toHashSet*[A](s: OrderedSet[A]): HashSet[A] =
```

Comment:
this should be
```nim
proc toHashSet*[A](s: IterableLen[A]): HashSet[A]
```
and merged with `proc toHashSet*[A](keys: openArray[A]): HashSet[A] =`
because all you're using is that it's iterable and has a `len`
=> needs a new concept `IterableLen` which is generally useful here and in sequtils among others

Reply:
Fixed. I also started a concepts module with the first two concepts used in the `sets` module: `Iterable[T]` and `IterableLen[T]`.